Server/src/main/java/org/gluu/oxauth/model/registration/RegisterParamsValidator.java

```diff
@@ -90,6 +92,94 @@
+                        registerRequest.getUserInfoEncryptedResponseAlg().toString()) ) {
+            log.debug("Parameter userinfo_encrypted_response_alg is not valid.");
+            return new Pair<>(false, "Parameter userinfo_encrypted_response_alg is not valid.");
+        }
+
+        if ( registerRequest.getUserInfoEncryptedResponseEnc() != null &&
+                ! appConfiguration.getUserInfoEncryptionEncValuesSupported().contains(
+                        registerRequest.getUserInfoEncryptedResponseEnc().toString()) ) {
+            log.debug("Parameter userinfo_encrypted_response_enc is not valid.");
+            return new Pair<>(false, "Parameter userinfo_encrypted_response_enc is not valid.");
+        }
+
+        if ( registerRequest.getRequestObjectSigningAlg() != null &&
+                ! appConfiguration.getRequestObjectSigningAlgValuesSupported().contains(
+                        registerRequest.getRequestObjectSigningAlg().toString()) ) {
+            log.debug("Parameter request_object_signing_alg is not valid.");
+            return new Pair<>(false, "Parameter request_object_signing_alg is not valid.");
+        }
+
+        if ( registerRequest.getRequestObjectEncryptionAlg() != null &&
+                ! appConfiguration.getRequestObjectEncryptionAlgValuesSupported().contains(
+                        registerRequest.getRequestObjectEncryptionAlg().toString()) ) {
+            log.debug("Parameter request_object_encryption_alg is not valid.");
+            return new Pair<>(false, "Parameter request_object_encryption_alg is not valid.");
+        }
+
+        if ( registerRequest.getRequestObjectEncryptionEnc() != null &&
+                ! appConfiguration.getRequestObjectEncryptionEncValuesSupported().contains(
+                        registerRequest.getRequestObjectEncryptionEnc().toString()) ) {
+            log.debug("Parameter request_object_encryption_enc is not valid.");
+            return new Pair<>(false, "Parameter request_object_encryption_enc is not valid.");
+        }
+
+        if ( registerRequest.getTokenEndpointAuthMethod() != null &&
+                ! appConfiguration.getTokenEndpointAuthMethodsSupported().contains(
+                        registerRequest.getTokenEndpointAuthMethod().toString()) ) {
+            log.debug("Parameter token_endpoint_auth_method is not valid.");
+            return new Pair<>(false, "Parameter token_endpoint_auth_method is not valid.");
+        }
+
+        if ( registerRequest.getTokenEndpointAuthSigningAlg() != null &&
+                ! appConfiguration.getTokenEndpointAuthSigningAlgValuesSupported().contains(
+                        registerRequest.getTokenEndpointAuthSigningAlg().toString()) ) {
+            log.debug("Parameter token_endpoint_auth_signing_alg is not valid.");
+            return new Pair<>(false, "Parameter token_endpoint_auth_signing_alg is not valid.");
+        }
+
+        return new Pair<>(true, "");
```

Comment:
Instead of returning result just throw error exception `throw errorResponseFactory.createWebApplicationException(Response.Status.BAD_REQUEST, RegisterErrorResponseType.INVALID_CLIENT_METADATA, message);`. Then all validation code will be only in validator and code will be more compact.